src/Snippets/Docs/Telemetry.cs

```diff
@@ -134,4 +134,36 @@ internal sealed class CustomMeteringEnricher : MeteringEnricher
     }
 
     #endregion
+
+    public static void SeverityOverrides()
+    {
+        var services = new ServiceCollection();
+
+        #region telemetry-severity-override
+
+        services.AddResiliencePipeline("my-strategy", (builder, context) =>
+        {
+            // Create a new instance of telemetry options by using copy-constructor of the global ones.
+            // This ensures that common configuration is preserved.
+            var telemetryOptions = new TelemetryOptions(context.GetOptions<TelemetryOptions>());
+
+            telemetryOptions.SeverityProvider = ev =>
+            {
+                if (ev.EventName == "OnRetry")
```

Comment:
I think if we would expose all the telemetry event name constants via a simple static class then the branching logic would safer. It would also help users of the API to use switch expression.

Reply:
It would require to expose a lot of public members for a limited audience. Let's wait and see. My vote would be to not expose anything unless absolutely necessary. Consumers of this API can just hardcode the strings, these won't change anyway as they are considered public contract as part of telemetry.

Reply:
In this specific case you could also do `nameof(RetryStrategyOptions<>.OnRetry)` (I think).

Reply:
The `nameof` won't work for all cases. For example 
- in case of rate limiter `OnRateLimiterRejected` event name vs `OnRejected` options property
- in case of circuit breaker `OnCircuitHalfOpened` event name vs `OnHalfOpened` options property
- in case of latency `Chaos.OnLatency` event name vs `OnLatencyInjected` options property
- etc.

Reply:
> It would require to expose a lot of public members for a limited audience. Let's wait and see. My vote would be to not expose anything unless absolutely necessary. Consumers of this API can just hardcode the strings, these won't change anyway as they are considered public contract as part of telemetry.

If you are saying that it is already implicitly part of the public contract then I don't see why it is a problem to make it explicit?

Reply:
Just more public area to cover :)